src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -931,29 +661,36 @@ protected function getCategoryToolbarButtons(Request $request)
     }
 
     /**
-     * Handle exception which occurs when deleting category image (cover, thumbnails).
-     *
-     * @param CategoryException $e
+     * Get translated error messsages for category exceptions
      *
-     * @return string
+     * @return array
      */
-    private function handleImageDeletingException(CategoryException $e)
+    private function getErrorMessages()
```

Comment:
this shouldn't be in the controller

Reply:
we already had this discussion a few months ago. :smile:

Reply:
so, it's sad to see it's still here /c @matks 

Let's say it simple: the only responsibility of a controller is to read a request and return a response.

Error Management should be extracted from the controllers, but more **important**: if controllers had less responsibilities / actions, we didn't even have to talk about it!

This is **because** you have too much actions in this controller that you were forced to create a private function to manage error messages :)